workers/tasks/import/assetStore.ts

```diff
@@ -19,55 +17,34 @@ export const generateAssetKeyForFile = (filePath) => {
 	return `${folderName}/${randomness}${now}.${fileExtension}`;
 };
 
-const blockForAssetFile = (assetKey) =>
-	new Promise((resolve, reject) => {
-		let attempts = 0;
-		const checkForFile = () => {
-			// @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
-			s3bucket
-				.headObject({ Key: assetKey })
-				.promise()
-				.then(resolve)
-				.catch(() => {
-					if (attempts > 5) {
-						return reject();
-					}
-					attempts += 1;
-					return setTimeout(() => {
-						checkForFile();
-					}, 1000);
-				});
-		};
-		checkForFile();
+const blockForAssetFile = (assetKey: string) =>
+	new Promise<void>(async (resolve, reject) => {
+		for (let i = 0; i < 5; i++) {
+			// eslint-disable-next-line no-await-in-loop
+			const exists = await assetsClient.checkIfFileExists(assetKey);
+			if (exists) {
+				return resolve();
+			}
+			// eslint-disable-next-line no-await-in-loop
+			await sleep(1000);
+		}
+		return reject();
 	});
 
 export const downloadFileFromAssetStore = (assetKey: string, filePath: string) =>
 	new Promise(async (resolve, reject) => {
 		await blockForAssetFile(assetKey);
```

Comment:
The refactor looks great to me overall, but I don't understand what `blockForAssetFile()` does. Looks like it deliberately slows things down if the asset file in question doesn't exist or else if the assets client takes a long time to check for the presence of the file. But why would it take a long time? And why would you only want to slow it down, not error out, or some other more complete shut-down?